### ISSUE TITLE
NLT Deleting Active Links in Table

### ISSUE THREAD

**Issue by brimwats** [trey-wallis/obsidian-notion-like-tables#248]
### Describe the bug

NLT is removes links in source, LP, and reading mode

### Steps to reproduce

Have a table like this 

![image](https://user-images.githubusercontent.com/73113656/188346718-9ba62cc9-1c41-4972-aeb5-aa8f25cc8df8.png)

add a block reference to it as required
![image](https://user-images.githubusercontent.com/73113656/188346722-01726fe2-2698-48ab-a88c-9222b7170807.png)

reload the app.
the NLT tables will work. add a new column, like date:
![image](https://user-images.githubusercontent.com/73113656/188347065-616c1331-8bfb-4a78-b016-d810bf13016d.png)

switch to LP: links are gone. in source too. The only links preserved are links to uncreated notes. Created notes will have their links deleted:

![image](https://user-images.githubusercontent.com/73113656/188347395-1d211f71-fec5-43f3-8c43-000dc40b088a.png)
![image](https://user-images.githubusercontent.com/73113656/188347505-9e95220a-2cb3-4b00-9d20-e24e4d4802c8.png)

### Expected behavior

No data loss. 

### Are you using the mobile app?

No

### Obsidian debug info

SYSTEM INFO:
	Obsidian version: v0.16.2
	Installer version: v0.15.6
	Operating system: Windows 10 Education 10.0.22000
	Login status: logged in
	Catalyst license: insider
	Insider build toggle: on
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: Minimal
	Snippets enabled: 24
	Restricted mode: off
	Plugins installed: 95
	Plugins enabled: 72
		1: Divide & Conquer v0.4.0
		2: Contextual Typography v2.2.4
		3: Advanced URI v1.24.0
		4: Another Quick Switcher v6.3.2
		5: AidenLx's Folder Note - folderv component v1.0.0
		6: AidenLx's Folder Note v0.16.4
		7: Folder Note Core v1.3.5
		8: File Explorer Note Count v1.2.0
		9: Better Command Palette v0.13.0
		10: Obsidian42 - BRAT v0.6.35
		11: Code Editor Shortcuts v1.10.0
		12: Zotero Integration v2.2.24
		13: Core Search Assistant v0.9.1
		14: LanguageTool Integration v0.3.0
		15: Hotkey Helper v0.3.15
		16: Custom Frames v2.4.3
		17: Sentence Navigator v1.0.1
		18: Reset Font Size v1.0.2
		19: Recent Files v1.3.3
		20: QuickAdd v0.5.5
		21: Plugin Changelogs v1.3.0
		22: Periodic Notes v0.0.17
		23: Paste image rename v1.5.0
		24: Outliner v3.0.0
		25: File path to URI v1.4.1
		26: Grandfather v1.0.1
		27: List Modified v1.1.2
		28: Longform v1.1.0
		29: Map View v2.1.1
		30: MetaEdit v1.7.2
		31: Note Auto Creator v1.2.0
		32: Obsidian Charts v3.6.3
		33: Obsidian Enhancing Export v1.1.4
		34: Open with v0.3.1
		35: Local File Interface v0.1.0
		36: Local images v0.14.2
		37: Omnisearch v1.5.2
		38: Customizable Menu v2.2.0
		39: Various Complements v7.2.2
		40: Pane Relief v0.3.4
		41: Metadata Menu v0.1.31
		42: Advanced Tables v0.17.3
		43: Global Hotkeys v0.1.2
		44: Supercharged Links v0.8.0
		45: Customizable Page Header and Title Bar v4.7.2
		46: Word Sprint v0.2.3
		47: Wikipedia v1.0.3
		48: Waypoint v1.3.0
		49: Virtual Hotkey Keyboard v1.1.3
		50: Upcoming v0.3.0
		51: Tweet to Markdown v2.10.3
		52: Templater v1.13.0
		53: Table Generator v1.1.1
		54: Style Settings v0.4.11
		55: Smarter Markdown Hotkeys v1.16.2
		56: Dataview v0.5.43
		57: DB Folder v2.3.4
		58: Dynamic Highlights v0.3.1
		59: Dynamic Table of Contents v0.0.27
		60: Excalidraw v1.7.19
		61: File Info Panel v1.2.5
		62: Footnote Shortcut v0.0.9
		63: Format Hotkeys v0.1.7
		64: Hider v1.2.1
		65: Hover Editor v0.11.4
		66: Minimal Theme Settings v6.0.2
		67: MySnippets v1.2.3
		68: Banners v1.3.3
		69: Quick Explorer v0.1.41
		70: Settings Search v1.2.0
		71: Keyboard Analyzer v0.1.1
		72: Notion-Like Tables v4.3.1

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

### Relevant log output

```shell
plugin:notion-like-tables:59802 
plugin:notion-like-tables:59803 loadAppData()
plugin:notion-like-tables:59804 Parsing table from element
plugin:notion-like-tables:59805 parsed Array(8)
plugin:notion-like-tables:59878 Loading data from cache.
plugin:notion-like-tables:59879 data Object
plugin:pane-relief:1 Uncaught ReferenceError: WorkspaceTabs is not defined
    at eval (plugin:pane-relief:1:18412)
    at Array.forEach (<anonymous>)
    at eval (plugin:pane-relief:1:18228)
index.ts:153 Dataview: all 3911 files have been indexed in 5.035s (3911 cached, 0 skipped).
fec-main.ts:43 <div class=​"nav-folder-title-content">​booxsidian​</div>​ styles empty
eval @ fec-main.ts:43
node:internal/errors:464 Uncaught (in promise) TypeError: The "path" argument must be of type string. Received null
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at validateString (node:internal/validators:119:11)
    at Object.join (node:path:429:7)
    at e.getFullRealPath (app.js:1:1070416)
    at e.getFullPath (app.js:1:1070339)
    at app.js:1:1062060
    at n (app.js:1:1070742)
fec-main.ts:43 <div class=​"nav-folder-title-content">​booxsidian​</div>​ styles empty
eval @ fec-main.ts:43
plugin:notion-like-tables:59112 
plugin:notion-like-tables:59113 saveAppData()
plugin:notion-like-tables:59114 new table markdown Object
plugin:notion-like-tables:59123 updated file content Object
plugin:notion-like-tables:59146 Updating settings cache
plugin:notion-like-tables:59147 data Object
main.ts:156 Uncaught (in promise) TypeError: checkbox.setChecked is not a function
    at eval (main.ts:156:12)
    at e.addToggle (app.js:1:1111205)
    at KeyboardAnalyzerSettingTab.display (main.ts:154:8)
    at SettingsSearch.getTabResources (plugin:settings-search:170:15)
    at SettingsSearch.buildPluginResources (plugin:settings-search:120:12)
    at eval (plugin:settings-search:122:29)
plugin:notion-like-tables:59802 
plugin:notion-like-tables:59803 loadAppData()
plugin:notion-like-tables:59804 Parsing table from element
plugin:notion-like-tables:59805 parsed Array(8)
plugin:notion-like-tables:59878 Loading data from cache.
plugin:notion-like-tables:59879 data Object
plugin:obsidian-list-modfied:745 t
cssBuilder.ts:244 Uncaught (in promise) TypeError: customCss.readCssFolders is not a function
    at eval (cssBuilder.ts:244:19)
    at Generator.next (<anonymous>)
    at fulfilled (tslib.es6.js:73:58)
plugin:notion-like-tables:59112 
plugin:notion-like-tables:59113 saveAppData()
plugin:notion-like-tables:59114 new table markdown Object
plugin:notion-like-tables:59123 updated file content Object
plugin:notion-like-tables:59146 Updating settings cache
plugin:notion-like-tables:59147 data Object
plugin:obsidian-list-modfied:745 t
plugin:notion-like-tables:59112 
plugin:notion-like-tables:59113 saveAppData()
plugin:notion-like-tables:59114 new table markdown Object
plugin:notion-like-tables:59123 updated file content Object
plugin:notion-like-tables:59146 Updating settings cache
plugin:notion-like-tables:59147 data Object
plugin:notion-like-tables:59802 
plugin:notion-like-tables:59803 loadAppData()
plugin:notion-like-tables:59804 Parsing table from element
plugin:notion-like-tables:59805 parsed Array(8)
plugin:notion-like-tables:59878 Loading data from cache.
plugin:notion-like-tables:59879 data Object
```

**Comment by trey-wallis** [comment]
@brimwats Thanks for bug report. Live preview is currently disabled - for me it doesn't even render a Notion-like table. I do notice that there are problems for me clicking any links in a table. I cannot reproduce the behavior of links being deleted though

**Comment by brimwats** [comment]
Sorry if I wasn't clear. I put it into reading mode to add a new column then switched back to LP. I notice now that all the links with supercharged links were deleted.

**Comment by trey-wallis** [comment]
Can you temporarily disable `supercharged-links` and repeat the same process and see if the behavior is the same?

**Comment by brimwats** [comment]
It still deletes them. It doesn't erase them entirely this time however.

A recording
![Obsidian_UryivvF4yC](https://user-images.githubusercontent.com/73113656/188360422-b0005842-a72d-4a3d-b1f0-c817ddc3dce3.gif)


So to summarize:

With NLT + Supercharged links→ `| [[2022-09-05\|Monday (09/05)]]|` is turned into `|Monday (09/05)|`

With just NLT → `| [[2022-09-05\|Monday (09/05)]]|` is turned into `| [[Monday (09/05)]]|`

**Comment by brimwats** [comment]
Ideally `| [[2022-09-05\|Monday (09/05)]]    |` would remain `| [[2022-09-05\|Monday (09/05)]]    |`

**Comment by brimwats** [comment]
logs

```
format-hotkeys-obsidian: 1ms
plugin:longform:16916 [Longform] Loaded and watching projects.
plugin:omnisearch:11 Omnisearch - indexing 3911 notes
4plugin:pane-relief:1 Uncaught ReferenceError: WorkspaceTabs is not defined
    at eval (plugin:pane-relief:1:18412)
    at Array.forEach (<anonymous>)
    at eval (plugin:pane-relief:1:18228)
plugin:notion-like-tables:59802 
plugin:notion-like-tables:59803 loadAppData()
plugin:notion-like-tables:59804 Parsing table from element
plugin:notion-like-tables:59805 parsed Array(8)
plugin:notion-like-tables:59878 Loading data from cache.
plugin:notion-like-tables:59879 data Object
plugin:pane-relief:1 Uncaught ReferenceError: WorkspaceTabs is not defined
    at eval (plugin:pane-relief:1:18412)
    at Array.forEach (<anonymous>)
    at eval (plugin:pane-relief:1:18228)
index.ts:153 Dataview: all 3911 files have been indexed in 5.25s (3911 cached, 0 skipped).
fec-main.ts:43 <div class=​"nav-folder-title-content">​booxsidian​</div>​ styles empty
eval @ fec-main.ts:43
node:internal/errors:464 Uncaught (in promise) TypeError: The "path" argument must be of type string. Received null
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at validateString (node:internal/validators:119:11)
    at Object.join (node:path:429:7)
    at e.getFullRealPath (app.js:1:1070416)
    at e.getFullPath (app.js:1:1070339)
    at app.js:1:1062060
    at n (app.js:1:1070742)
fec-main.ts:43 <div class=​"nav-folder-title-content">​booxsidian​</div>​ styles empty
eval @ fec-main.ts:43
main.ts:156 Uncaught (in promise) TypeError: checkbox.setChecked is not a function
    at eval (main.ts:156:12)
    at e.addToggle (app.js:1:1111205)
    at KeyboardAnalyzerSettingTab.display (main.ts:154:8)
    at SettingsSearch.getTabResources (plugin:settings-search:170:15)
    at SettingsSearch.buildPluginResources (plugin:settings-search:120:12)
    at eval (plugin:settings-search:122:29)
plugin:notion-like-tables:59112 
plugin:notion-like-tables:59113 saveAppData()
plugin:notion-like-tables:59114 new table markdown Object
plugin:notion-like-tables:59123 updated file content Object
plugin:notion-like-tables:59146 Updating settings cache
plugin:notion-like-tables:59147 data Object
plugin:obsidian-list-modfied:745 t
plugin:notion-like-tables:59802 
plugin:notion-like-tables:59803 loadAppData()
plugin:notion-like-tables:59804 Parsing table from element
plugin:notion-like-tables:59805 parsed Array(8)
plugin:notion-like-tables:59878 Loading data from cache.
plugin:notion-like-tables:59879 data Object
plugin:notion-like-tables:59112 
plugin:notion-like-tables:59113 saveAppData()
plugin:notion-like-tables:59114 new table markdown Object
plugin:notion-like-tables:59123 updated file content Object
plugin:notion-like-tables:59146 Updating settings cache
plugin:notion-like-tables:59147 data Object
plugin:obsidian-list-modfied:745 t
```

**Comment by trey-wallis** [comment]
So I disabled all plugins, restarted Obsidian, and tried using the alias feature. It seems like Obsidian tables themselves are broken?

<img width="542" alt="image" src="https://user-images.githubusercontent.com/40307803/188365178-f8a899ec-952d-40cd-ae16-684a6747e335.png">
<img width="382" alt="image" src="https://user-images.githubusercontent.com/40307803/188365217-2ab5eb14-3132-4f2a-9dea-0512bf0f25da.png">

Could you try this could as well?

```markdown
| Column 1 |
| --------  |
| [[Link|alias]] |

[[Link|alias]]
```

**Comment by brimwats** [comment]
you have to backslash before a alias in a table. it's always been that way

**Comment by trey-wallis** [comment]
I didnt know that haha.

**Comment by brimwats** [comment]
> I didnt know that haha.

no worries!

could it be NLTs regex is only picking up the alias?

**Comment by trey-wallis** [comment]
I think it was something like that. I didn't bother fixing it because I am just overhauling the entire system. It will fix all HTML entity errors and your table should appear identical.

**Comment by trey-wallis** [comment]
@brimwats Should be fixed in v5.0.0. if you have any other issues please open another issue. Thank you for your feedback and support :)
